### PR TITLE
Refactor/adjust project structure

### PR DESCRIPTION
**Pull request by FoRVaiS**
#### Routes
 - Create a centralized file for defining routes
#### Response
 - Create controller wrapper to automate response formatting/creation
#### Pagination
 - Introduce middleware to extract  `page` and `limit` query params and inject them into the request object
 - Add pagination links to response

### Testing
1. Spin up resources: `make res`
2. Start the api server: `npm run -w apps/api start:dev`

#### Verify routes still work
3. Use Postman (or Thunder Client) to login with dummy admin credentials. `http://127.0.0.1:3000/api/v1/auth/login`
    - admin@email.com: admin
4. Expect the following response:
```json
{
  "isError": false,
  "data": {
    "userId": 1
  },
  "pagination": {
    "prev": null,
    "next": null
  }
}
```

### CODE REVIEW

**Comment by FoRVaiS** [comment]
Hours Worked: 3

File: apps/api/src/server/server.ts [hunk 1]
```diff
@@ -1,29 +1,25 @@
+import { type Resources } from './services/Service';
+
 import express from 'express';
 
 // Middleware
 import helmet from 'helmet';
 import morgan from 'morgan';
+import { pagination } from './middleware/PaginationMiddleware';
```
Comment (tgrigoruk) [inline apps/api/src/server/server.ts:8]: Module is missing - cannot run project.

**Review by tgrigoruk — CHANGES_REQUESTED** [review]

File: apps/api/src/server/services/AuthService.ts [hunk 2]
```diff
@@ -65,84 +65,68 @@ export class AuthController {
   }
 }
 
-export class AuthService extends BaseService {
+export class AuthService {
```
Comment (tgrigoruk) [inline apps/api/src/server/services/AuthService.ts:68]: Is `BaseService` being deprecated? If so the import above (line 8) can be removed, and the class itself removed from `Service.ts`.

**Comment by tgrigoruk** [comment]
If i try a non existing route the response is `404` and contains
```json
"isError": false,
"data": "Route Not Found"
```
Is this intended behaviour? It could be a problem if one checks the response for `isError` rather than status code.

**Comment by FoRVaiS** [comment]
> If i try a non existing route the response is `404` and contains
> 
> ```json
> "isError": false,
> "data": "Route Not Found"
> ```
> 
> Is this intended behaviour? It could be a problem if one checks the response for `isError` rather than status code.

Not intended, I forgot to pass an error into the response function, fixed now.

**Review by tgrigoruk — APPROVED** [review]

**Pull request merged**